### PR TITLE
Feature/automaton

### PR DESCRIPTION
**Pull request by kant01ne**
Unfinished work on automaton/action and state

### CODE REVIEW

File: src/automaton.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
+//
+//  automaton.h
+//  lut-lang
+//
+//  Created by Kevin Antoine on 06/03/2015.
+//  Copyright (c) 2015 H4314. All rights reserved.
+//
+
+#include "state.h"
+#include Token.h"
```
Comment (rricard) [inline src/automaton.h]: "Token.h"

**Comment by kant01ne** [comment]
RTM @rricard

File: src/Automaton.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
+//
+//  automaton.h
+//  lut-lang
+//
+//  Created by Kevin Antoine on 06/03/2015.
+//  Copyright (c) 2015 H4314. All rights reserved.
+//
+#include <map>
+#include "src/State.h"
+#include "src/Token.h"
```
Comment (rricard) [inline src/Automaton.h]: Where is this ? Use `ASTTokenNode.h`

File: .gitignore [hunk 3]
```diff
@@ -17,7 +17,7 @@
 *.mod
 
 # Compiled Static libraries
-*.lai
+*.laiSZ
```
Comment (rricard) [inline .gitignore]: Why ?

File: README.md [hunk 4]
```diff
@@ -27,4 +27,53 @@ Usage
...
+Terminal symbols
+
+var : variable
+const : constant
+id : variable name
+val : positive value
+
+
+Non terminal symbols
+------
```
Comment (rricard) [inline README.md:40]: Go to the end of the line with dashes

File: README.md [hunk 5]
```diff
@@ -27,4 +27,53 @@ Usage
...
+L2 : EnumAssign
+I : InstructionBlock
+E : FirstLevelExpression
+T : SecondLevelExpression
+F : ThirdLevelExpression
+opA : AdditiveOperation
+opM : MultiplicativeOperation
+
+Grammar Rules
+-------
```
Comment (rricard) [inline README.md:54]: Same here

File: src/Automaton.h [hunk 6]
```diff
@@ -0,0 +1,31 @@
+//
+//  automaton.h
+//  lut-lang
+//
+//  Created by Kevin Antoine on 06/03/2015.
+//  Copyright (c) 2015 H4314. All rights reserved.
+//
+#include <map>
+#include "State.h"
+#include "AstTokenNode.h"
```
Comment (rricard) [inline src/Automaton.h]: **AST**

**Comment by rricard** [comment]
Add your files to makefile

**Comment by kant01ne** [comment]
RTM @rricard

**Comment by rricard** [comment]
Checking this later

**Pull request merged**